presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -418,9 +422,12 @@ public static LocalDateTime fromPrestoTimestamp(long value)
                 int decimalDigits = type.getDecimalDigits();
                 int precision = columnSize + max(-decimalDigits, 0); // Map decimal(p, -s) (negative scale) to decimal(p+s, 0).
                 if (precision > Decimals.MAX_PRECISION) {
-                    return Optional.empty();
+                    if (getDecimalRoundingMode(session) == UNNECESSARY) {
+                        return Optional.empty();
```

Comment:
I would keep processing and fail at runtime when actual value does not fit the type and would require rounding.

Reply:
It might break queries using `select * from table` unless we have https://github.com/prestosql/presto/pull/1182
For example table with a column of `decimal(50,0)`

Reply:
I think we could map `decimal(50,0)` to `decimal(38,0)` as long as values fit into it.

Reply:
Make sense if it won't release before #1182

Reply:
> I think we could map `decimal(50,0)` to `decimal(38,0)` as long as values fit into it.
@kokosing 
This mapping doesn't need rounding since scale is known, so I skipped it.

Reply:
Please `break`, instead of `return Optional.empty()`.